src/HDF5.jl

```diff
@@ -1535,11 +1563,11 @@ function readmmap(obj::HDF5Dataset, ::Type{Array{T}}) where {T}
     end
     if !Sys.iswindows()
         local fdint
-        prop = h5d_get_access_plist(checkvalid(obj).id)
+        prop = h5d_get_access_plist(obj.id)
```

Comment:
I removed this use of `checkvalid` here because calling `size(obj)` already checks for validity.